packaging/specifiers.py

```diff
@@ -965,7 +965,7 @@ def filter(
         if self._specs:
             for spec in self._specs:
                 iterable = spec.filter(iterable, prereleases=bool(prereleases))
```

Comment:
This loop looks weird. Is it essentially running `reduce` on itself? And how do multiple calls lead to a different outcome?